From 7b529aad973bef69611924788e36b85f8b844290 Mon Sep 17 00:00:00 2001
From: "Alan T. DeKok" <aland@freeradius.org>
Date: Sat, 21 Dec 2024 06:49:54 -0500
Subject: [PATCH] EAP-TEAP: Don't complain about missing PAC when
 teap_provisioning=0

If we're not provisioning, then we don't need the PAC.

RFC 7170bis officially deprecates the PAC, and it doesn't
appear that other TEAP implementations use it.

Signed-off-by: Alan DeKok <aland@freeradius.org>
---
 src/eap_peer/eap_teap.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/eap_peer/eap_teap.c b/src/eap_peer/eap_teap.c
index ced7b1642..9a6c2dff4 100644
--- a/src/eap_peer/eap_teap.c
+++ b/src/eap_peer/eap_teap.c
@@ -205,6 +205,9 @@ static void * eap_teap_init(struct eap_sm *sm)
 		return NULL;
 	}
 
+	if (!data->provisioning_allowed && !config->pac_file)
+		return data;
+
 	if (!config->pac_file) {
 		wpa_printf(MSG_INFO, "EAP-TEAP: No PAC file configured");
 		eap_teap_deinit(sm, data);
-- 
2.43.0

